packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -149,11 +159,31 @@ export class DocumentDeltaConnection
             // though some logic (naming assert in initialMessages getter) might need to be adjusted (it becomes noop)
             assert((this.listeners(event).length !== 0) === this.trackedListeners.has(event), 0x20b /* "mismatch" */);
             if (!this.trackedListeners.has(event)) {
-                this.addTrackedListener(
-                    event,
-                    (...args: any[]) => {
-                        this.emit(event, ...args);
-                    });
+                let listener: (...args: any[]) => void;
+                switch (event) {
+                    case "op": {
+                        listener = (docId: string, messages: ISequencedDocumentMessage[]) => {
+                            if (!this.lightOpAcksEnabled) {
+                                this.emit(event, docId, messages);
+                            } else {
+                                this.emit(event, docId, messages.map((message) => {
+                                    // Repopulate op contents if applicable
+                                    if (this.clientId === message.clientId && this.opContent.has(message.clientSequenceNumber)) {
```

Comment:
Not sure if we will need it, but you can make it a bit more prepared for potential future optimization (not sure even if they are possible, but it does not make things worse, I think) and check not clientId, but docId.
That way, maybe, we will be able to make it work for case of interactive client + summarizer client on same socket in the future.

Reply:
I believe docId here is PUSH docId, and it uniquely identifies document for this connection. Can you please poke around? I believe there are other places where we do comparisons with it, so hopefully it becomes clearer on what it tracks - you can always ask Gary, but probably we want to follow up with AFR folks as well.

Reply:
In the example I was running the socket/document ID was the same across different client IDs, so I think we have to stick with clientId for now